crates/matrix-sdk/src/client/login_builder.rs

```diff
@@ -0,0 +1,313 @@
+#![cfg_attr(not(target_arch = "wasm32"), deny(clippy::future_not_send))]
+
+#[cfg(all(feature = "sso-login", not(target_arch = "wasm32")))]
+use std::future::Future;
+
+use ruma::{
+    api::client::{session::login, uiaa::UserIdentifier},
+    assign,
+};
+use tracing::{info, instrument};
+
+use super::Client;
+use crate::{config::RequestConfig, Result};
+
+enum LoginMethod<'a> {
+    UserPassword { id: UserIdentifier<'a>, password: &'a str },
+    Token(&'a str),
```

Comment:
I adminit it's kind of obvious, but having documentation here won't be superfluous, esp. for the `Token` variant. Maybe adding a link to the spec to precise what's the form of the token etc., would be useful.

Reply:
Sure, I'll add that. Note that it's not going to be visible in documentation by default because this enum is private. I'll find another location to also link to the spec in the public docs.

Reply:
Actually I'm not so sure about needing a link to the spec in the public docs, I don't think we link to the spec for other endpoints.